executor/executor_test.go

```diff
@@ -1888,19 +1884,33 @@ func (s *testSuite) TestPointGetRepeatableRead(c *C) {
 	tk2 := testkit.NewTestKit(c, s.store)
 	tk2.MustExec("use test")
 
+	var (
+		step1 = "github.com/pingcap/tidb/executor/pointGetRepeatableReadTest-step1"
+		step2 = "github.com/pingcap/tidb/executor/pointGetRepeatableReadTest-step2"
+	)
+
+	c.Assert(failpoint.Enable(step1, "return"), IsNil)
+	c.Assert(failpoint.Enable(step2, "pause"), IsNil)
+
+	updateWaitCh := make(chan struct{})
 	go func() {
-		ctx := context.WithValue(context.Background(), "pointGetRepeatableReadTest", true)
+		ctx := context.WithValue(context.Background(), "pointGetRepeatableReadTest", updateWaitCh)
+		ctx = failpoint.WithHook(ctx, func(ctx context.Context, fpname string) bool {
+			return fpname == step1 || fpname == step2
+		})
 		rs, err := tk1.Se.Execute(ctx, "select c from point_get where b = 1")
 		c.Assert(err, IsNil)
 		result := tk1.ResultSetToResultWithCtx(ctx, rs[0], Commentf("execute sql fail"))
 		result.Check(testkit.Rows("1"))
 	}()
 
-	label := debugger.Bind("point-get-g2")
-	debugger.Continue("point-get-g1")
-	debugger.Breakpoint(label)
+	defer func() {
+		c.Assert(failpoint.Disable(step1), IsNil)
+		c.Assert(failpoint.Disable(step2), IsNil)
+	}()
+
+	<-updateWaitCh // Wait `POINT GET` first time `get`
 	tk2.MustExec("update point_get set b = 2, c = 2 where a = 1")
```

Comment:
`failpoint.Disable(step2)` should be here?

Reply:
After this line?

Reply:
Yes, `step2` means point get wait for `UPDATE` finish and defer will execute after L1913.

Reply:
I have reorganized the part, PTAL again.